web/docs/use-vast/detect/match-threat-intel.md

```diff
@@ -0,0 +1,553 @@
+utility `bloom`:
+
+```bash
+bloom create -p 0.1 -n 100 ns.bloom
+echo 1.1.1.1,8.8.8.8 | bloom -s insert ns.bloom
+```
+
+Use `bloom show ns.bloom` to display a few statistics about the filter, such as
+the number of elements, false-positive probability, number of hash functions,
+and bits used.
+
+Finally, we hand the Bloom filter over to VAST and associate it with the
+matcher called `ns`:
+
+```sh
+vast matcher load dns < ns.bloom
+```
+
+See the section on [matcher state management](#manage-matcher-state) for
+a more detailed discussion on loading binary state into the matcher.
+
+## Examples
+
+This section includes real-world examples to illustrate how the matcher works
+in practice.
+
+### IP Blocklists
+
+IP blocklists make up for a large share of low-level IoCs, often to represent
+attacker infrastructure, such as C2 servers.
+
+The [Feodo Tracker](https://feodotracker.abuse.ch/) from
+[abuse.ch](https://abuse.ch) represents one such blocklist that gets updated
+every 5 minutes. Let's take a look:
+
+```bash
+head -n 15 ipblocklist.csv
+################################################################
+# abuse.ch Feodo Tracker Botnet C2 IP Blocklist (CSV)          #
+# Last updated: 2021-08-17 15:00:42 UTC                        #
+#                                                              #
+# Terms Of Use: https://feodotracker.abuse.ch/blocklist/       #
+# For questions please contact feodotracker [at] abuse.ch      #
+################################################################
+#
+"first_seen_utc","dst_ip","dst_port","c2_status","last_online","malware"
+"2021-01-17 07:30:05","67.213.75.205","443","offline","2021-08-18","Dridex"
+"2021-01-17 07:44:46","51.178.161.32","4643","online","2021-08-18","Dridex"
+"2021-01-17 07:44:50","162.144.127.197","3786","online","2021-08-18","Dridex"
+"2021-01-17 07:45:55","111.230.104.169","3388","online","2021-08-18","Dridex"
+"2021-01-17 07:45:58","217.79.184.243","33443","online","2021-08-18","Dridex"
+"2021-01-17 07:47:59","46.101.90.205","4643","online","2021-08-18","Dridex"
+```
+
+Before VAST can read this data, we need to tell VAST what type to use for it.
+
+```
+type abuse.feodo.blocklist = record {
+  first_seen_utc: time,
+  dst_ip: addr,
+  dst_port: port,
+  c2_status: enum { online, offline },
+  last_online: time,
+  malware: string,
+}
+```
+
+In addition, you need to tell VAST what fields have the indicator data,
+consisting of value and an optional context. To this end, you need to provide a
+concept definition:
+
+```yaml
+- concept:
+    name: matcher.indicator.value
+    fields:
+      - abuse.feodo.blocklist.dst_ip
+- concept:
+    name: matcher.indicator.context
+    fields:
+      - abuse.feodo.blocklist.malware
+```
+
+:::note Upcoming changes
+We understand that this specification feels cumbersome and are working on a
+solution. In the future, you'll be able to define type and concepts in a single
+YAML file. Stay tuned.
+:::
+
```

Comment:
This can probably be removed already given that the docs already refer to the YAML syntax.